go/src/koding/db/mongodb/modelhelper/group.go

```diff
@@ -58,6 +60,35 @@ func GetGroupFieldsByIds(ids []string, fields []string) ([]*models.Group, error)
 	return groups, Mongo.Run(GroupsCollectionName, query)
 }
 
+// GetGroupForKite reverse looks up a team name for the given kiteID
+// by looking up a kiteID among jMachine.queryString fields.
+func GetGroupForKite(kiteID string) (*models.Group, error) {
+	qs, err := utils.QueryString(kiteID)
+	if err != nil {
+		return nil, err
+	}
+
+	var groups struct {
+		Groups []struct {
+			ID bson.ObjectId `bson:"id"`
+		} `bson:"groups"`
+	}
+
+	fn := func(c *mgo.Collection) error {
+		return c.Find(bson.M{"queryString": qs}).Select(bson.M{"groups": 1}).One(&groups)
```

Comment:
we dont have index on "queryString", we should find another way to achieve this temporarily